x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/monitor_status/monitor_status_header.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiTitle,
+  EuiLink,
+} from '@elastic/eui';
+import { css } from '@emotion/css';
+import { useHistory } from 'react-router-dom';
+
+import { MONITOR_HISTORY_ROUTE } from '../../../../../../common/constants';
+import { stringifyUrlParams } from '../../../utils/url_params';
+import { useGetUrlParams } from '../../../hooks';
+
+import { useSelectedMonitor } from '../hooks/use_selected_monitor';
+
+import * as labels from './labels';
+import { MonitorStatusPanelProps } from './monitor_status_data';
+
+export const MonitorStatusHeader = ({
+  from,
+  to,
+  periodCaption,
+  showViewHistoryButton,
+}: MonitorStatusPanelProps) => {
+  const history = useHistory();
+  const params = useGetUrlParams();
+  const { monitor } = useSelectedMonitor();
+
+  const isLast24Hours = from === 'now-24h' && to === 'now';
+  const periodCaptionText = !!periodCaption
+    ? periodCaption
+    : isLast24Hours
+    ? labels.LAST_24_HOURS_LABEL
+    : '';
+
+  return (
+    <EuiFlexGroup
+      direction="row"
+      alignItems="baseline"
+      css={css`
+        margin-bottom: 0;
+      `}
+    >
+      <EuiFlexItem grow={false}>
+        <EuiTitle size="xs">
+          <h4>{labels.STATUS_LABEL}</h4>
+        </EuiTitle>
+      </EuiFlexItem>
+      {periodCaptionText ? (
+        <EuiFlexItem grow={false}>
+          <EuiText size="xs" color="subdued">
+            {periodCaptionText}
+          </EuiText>
+        </EuiFlexItem>
+      ) : null}
+      <EuiFlexItem grow={true} />
+
+      {showViewHistoryButton ? (
+        <EuiFlexItem grow={false}>
+          <EuiLink
+            href={
+              monitor?.id
+                ? history.createHref({
+                    pathname: MONITOR_HISTORY_ROUTE.replace(':monitorId', monitor?.id),
+                    search: stringifyUrlParams(
+                      { ...params, dateRangeStart: 'now-24h', dateRangeEnd: 'now' },
+                      true
+                    ),
+                  })
+                : undefined
+            }
+          >
+            <EuiButtonEmpty
+              data-test-subj="monitorStatusChartViewHistoryButton"
+              size="xs"
+              iconType="list"
+            >
+              {labels.VIEW_HISTORY_LABEL}
+            </EuiButtonEmpty>
+          </EuiLink>
+        </EuiFlexItem>
```

Comment:
It might be cleaner to use https://github.com/elastic/kibana/blob/main/x-pack/plugins/synthetics/public/apps/synthetics/components/common/react_router_helpers/link_for_eui.tsx here